packages/design-system/src/components/LinkButton/LinkButton.tsx

```diff
@@ -1,90 +1,30 @@
 import * as React from 'react';
 
-import { styled } from 'styled-components';
-
-import { focus } from '../../styles/buttons';
 import { PolymorphicRef } from '../../types';
 import { forwardRef } from '../../utilities/forwardRef';
-import { BaseLink, BaseLinkComponent, BaseLinkProps } from '../BaseLink';
-import { VARIANTS, ButtonSize, ButtonVariant } from '../Button/constants';
-import { getDisabledStyle, getHoverStyle, getActiveStyle, getVariantStyle } from '../Button/utils';
-import { Flex } from '../Flex';
-import { Typography } from '../Typography';
+import { BaseLink } from '../BaseLink';
+import { Button, ButtonProps } from '../Button';
 
-type LinkButtonProps<C extends React.ElementType = 'a'> = BaseLinkProps<C> & {
-  disabled?: boolean;
-  endIcon?: React.ReactNode;
-  size?: ButtonSize;
-  startIcon?: React.ReactNode;
-  variant?: ButtonVariant;
-};
+type LinkButtonProps<C extends React.ElementType = typeof BaseLink> = ButtonProps<C>;
```

Comment:
this is not clear, 

why this is not `type LinkButtonProps<C extends React.ElementType = "button"> = ButtonProps<C>;`

how this LinkButton wrapped exactly? even this syntax is confusing for me, `<Button<typeof BaseLink>`. If you could please elaborate or discuss on call.

Reply:
Because the `tag` component used in the `Button` component below is `BaseLink`. In our polymorphic components, the C generic denotes the element which corresponds directly to the `tag` prop. 

The `<Button<typeof BaseLink>` is because all components are just functions, and the `Button` component is a generic function that takes it's element type. So in this scenario because we're working with two layers of polymoprhism we add the component as the generic that our props correspond too to avoid type errors.

Reply:
ohk, I will check how this polymorphism works here, thanks